cmd/server/main.go

```diff
@@ -1,3 +1,107 @@
 package main
 
-func main() {}
+import (
+	"flag"
+	"fmt"
+	"net/http"
+	"os"
+	"os/signal"
+	"time"
+
+	service "github.com/vilasle/metrics/internal/service/server"
+
+	"github.com/vilasle/metrics/internal/repository/memory"
+	rest "github.com/vilasle/metrics/internal/transport/rest/server"
+)
+
+type runConfig struct {
+	address string
+}
+
+func getConfig() runConfig {
+	address := flag.String("a", "localhost:8080", "address for server")
+
+	flag.Parse()
+
+	envAddress := os.Getenv("ADDRESS")
+	if envAddress != "" {
+		*address = envAddress
+	}
+
+	return runConfig{
+		address: *address,
+	}
+}
+
+func main() {
+	defer func() {
+		if err := recover(); err != nil {
+			fmt.Println("application is in a panic", err)
+		}
+	}()
+
+	conf := getConfig()
+
+	gaugeStorage := memory.NewMetricGaugeMemoryRepository()
+	counterStorage := memory.NewMetricCounterMemoryRepository()
+
+	svc := service.NewStorageService(gaugeStorage, counterStorage)
+
+	server := rest.NewHTTPServer(conf.address)
+
+	server.Register("/", methods(), contentTypes(), rest.DisplayAllMetrics(svc))
+	server.Register("/value/", methods(http.MethodGet), contentTypes(), rest.DisplayMetric(svc))
```

Comment:
```suggestion
	server.Register("/value/{type}/{name}", methods(http.MethodGet), contentTypes(), rest.DisplayMetric(svc))
```
И дальше разбирать штатными методами.